modules/entity_version_history/tests/src/FunctionalJavascript/HistoryTabSettingsTest.php

```diff
@@ -0,0 +1,169 @@
+
+    // Check that there are only 3 config entities created.
+    $storage = $this->container->get('entity_type.manager')->getStorage('entity_version_history_settings');
+    $config_entities = $storage->loadMultiple();
+    $this->assertEquals(3, count($config_entities));
+
+    // Make sure that the settings are reflected in the form and all checked.
+    $this->assertTrue($node_entity_checkbox->isChecked());
+    $this->assertTrue($first_bundle_checkbox->isChecked());
+    $this->assertTrue($second_bundle_checkbox->isChecked());
+    $this->assertTrue($history_entity_checkbox->isChecked());
+    $this->assertTrue($history_bundle_checkbox->isChecked());
+
+    // Make sure configuration saved correctly and complies with the schema.
+    $config = $this->config('entity_version_history.settings.node.first_bundle');
+    $this->assertEquals('node', $config->get('target_entity_type_id'));
+    $this->assertEquals('first_bundle', $config->get('target_bundle'));
+    $this->assertEquals('field_entity_version', $config->get('target_field'));
+    $this->assertConfigSchema($this->container->get('config.typed'), $config->getName(), $config->get());
+
+    $config = $this->config('entity_version_history.settings.node.second_bundle');
+    $this->assertEquals('node', $config->get('target_entity_type_id'));
+    $this->assertEquals('second_bundle', $config->get('target_bundle'));
+    $this->assertEquals('field_entity_version', $config->get('target_field'));
+    $this->assertConfigSchema($this->container->get('config.typed'), $config->getName(), $config->get());
+
+    $config = $this->config('entity_version_history.settings.history_entity_test.history_entity_test');
+    $this->assertEquals('history_entity_test', $config->get('target_entity_type_id'));
+    $this->assertEquals('history_entity_test', $config->get('target_bundle'));
+    $this->assertEquals('version', $config->get('target_field'));
+    $this->assertConfigSchema($this->container->get('config.typed'), $config->getName(), $config->get());
+
+    // Remove configs by unchecking history entity checkbox and
+    // the first_bundle checkbox from node entity.
+    $history_entity_checkbox->uncheck();
+    $first_bundle_checkbox->uncheck();
+
+    $page->pressButton('Save configuration');
+
+    // Make sure that the settings are reflected in the form.
+    $this->assertTrue($node_entity_checkbox->isChecked());
+    $this->assertTrue($second_bundle_checkbox->isChecked());
+
+    $this->assertFalse($first_bundle_checkbox->isChecked());
+    $this->assertFalse($history_entity_checkbox->isChecked());
+    $this->assertFalse($history_bundle_checkbox->isChecked());
+
+    // Check the configs are deleted.
+    $this->container->get('config.factory')->clearStaticCache();
+    $config = $this->config('entity_version_history.settings.history_entity_test.history_entity_test');
+    $this->assertNull($config->get('target_entity_type_id'));
```

Comment:
Why are you asserting the deleted config like this? Just try to load it with the storage and see it doesnt find it. Like at like line 99.

Reply:
I tried everything I know to do it. For some reason cache is going on still because it loads them. I debugged manually and in the form the same code loads null, that is why our assertion for the checkboxes not being checked are true. I have tried with `$storage->resetCache();` and to use ::service() instead of the container as it is advised, no luck. As you can see after try the same from config() it works. If you have any idea please help me.